cmd/adapter/main.go

```diff
@@ -1,14 +1,109 @@
-package adapter
+package main
 
-// Build-time variables set via ldflags (reserved for future version/build info display)
+import (
+	"context"
+	"errors"
+	"flag"
+	"fmt"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	"github.com/cloudevents/sdk-go/v2/event"
+	"github.com/openshift-hyperfleet/hyperfleet-adapter/internal/broker_consumer"
+	"github.com/openshift-hyperfleet/hyperfleet-adapter/pkg/logger"
+)
+
+// Build-time variables set via ldflags
 var (
-	_ = version   // Unused: reserved for --version flag
-	_ = commit    // Unused: reserved for build info
-	_ = buildDate // Unused: reserved for build info
-	_ = tag       // Unused: reserved for build info
-
-	version   string
-	commit    string
-	buildDate string
-	tag       string
+	version   = "dev"
+	commit    = "none"
+	buildDate = "unknown"
+	tag       = "none"
 )
+
+const shutdownTimeout = 30 * time.Second
+
+func main() {
+	// Initialize glog flags
+	flag.Parse()
+	// Flush logs on exit
+	defer logger.Flush()
+
+	// Create context that cancels on system signals
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	// Create logger with context
+	log := logger.NewLogger(ctx)
+
+	log.Infof("Starting Hyperfleet Adapter version=%s commit=%s built=%s tag=%s", version, commit, buildDate, tag)
+
+	// Handle signals for graceful shutdown
+	sigCh := make(chan os.Signal, 1)
+	signal.Notify(sigCh, syscall.SIGINT, syscall.SIGTERM)
+	go func() {
+		sig := <-sigCh
+		log.Infof("Received signal %s, initiating graceful shutdown...", sig)
+		cancel()
+
+		// Second signal forces immediate exit
+		sig = <-sigCh
+		log.Infof("Received second signal %s, forcing immediate exit", sig)
+		os.Exit(1)
+	}()
+
+	// Create broker subscriber
+	// This will automatically read SUBSCRIPTION_ID and broker config from env vars/config file
+	subscriber, subscriptionID, err := broker_consumer.NewSubscriber("")
+	if err != nil {
+		log.Fatal(fmt.Sprintf("Failed to create subscriber: %v for subscription: %s", err, subscriptionID))
+	}
+	
+	defer func() {
+		// Use a timeout for closing to prevent hanging forever
+		closeCtx, closeCancel := context.WithTimeout(context.Background(), shutdownTimeout)
+		defer closeCancel()
+
+		done := make(chan struct{})
+		go func() {
+			subscriber.Close()
```

Comment:
If the NewSubscriber fails, will subscriber here be nil? Potential racey condition